testnet/stacks-node/src/neon_node.rs

```diff
@@ -600,8 +601,11 @@ fn spawn_miner_relayer(
                     }
                 }
                 RelayerDirective::RegisterKey(ref last_burn_block) => {
-                    debug!("Relayer: Register key");
-                    rotate_vrf_and_register(
+                    if did_register_key {
```

Comment:
The miner wouldn't know that the register key operation was accepted or not when `did_register_key` was set -- I think that would just cause the miner to never successfully mine, and could be remedied by restarting. Was this necessary because `RegisterKey` directives are being issued multiple times on the first block that the miner activates for (because the miner gets "woken up" multiple times per block now, unlike when the neon_node was first written)?

Reply:
>  Was this necessary because RegisterKey directives are being issued multiple times on the first block

Correct

Reply:
I can reset this flag, when the a directive is occurring on a different block height, but it still feels like a patch, from what I saw at runtime PoxWatchdog looks very "nervous" and could receive an upgrade.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/2073/commits/c1798bc165053d18c1c2ccf4275041237fa0b9b0